src/WebJobs.Script.WebHost/Security/Authentication/ScriptJwtBearerExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.Azure.Web.DataProtection;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.WebHost.Security.Authentication;
+using Microsoft.IdentityModel.Tokens;
+using static Microsoft.Azure.WebJobs.Script.EnvironmentSettingNames;
+using static Microsoft.Azure.WebJobs.Script.ScriptConstants;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ScriptJwtBearerExtensions
+    {
+        public static AuthenticationBuilder AddScriptJwtBearer(this AuthenticationBuilder builder)
+            => builder.AddJwtBearer(o =>
+                        {
+                            o.Events = new JwtBearerEvents()
+                            {
+                                OnTokenValidated = c =>
```

Comment:
I don't see any tests for this type. Need to factor this stuff out into testable functions and add tests.